### PR TITLE
Challenge - ES6: Understand the Differences Between import and require - Fixed test and added solution

### LINKED ISSUE #17436: ES6 - Understand the Differences Between import and require - correct syntax not passing
#### test not passing
in this challenge, the final "NOTE" provided suggests using "./" if the file containing functions to import it are in the same folder, however the solution using that hint does not pass. 

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/understand-the-differences-between-import-and-require/

#### Tell us about your browser and operating system

* Browser Name: Chrome
* Browser Version: Version 66.0.3359.181 (Official Build) (64-bit)
* Operating System: High Sierra 10.13.4

#### If possible, add a screenshot here (not a screenshot, but this code does not pass) and it should, and is 'more correct' than "string_functions" since many compilers will give an error without the "./"
```
"use strict";
import { capitalizeString } from "./string_functions"
capitalizeString("hello!");

```

### PR DESCRIPTION
**Pull request by alanpaulprice**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #17436

### CODE REVIEW

**Comment by alanpaulprice** [comment]
@ezioda004 do you want to include any changes in relation to #17866?

**Comment by ezioda004** [comment]
@alanpaulprice I'd say adding a Note for that would be a good idea. 
We should discuss the verbiage on the issue itself (so that others can chime in as well).

**Comment by alanpaulprice** [comment]
@ezioda004 Sure.

If I'm correct, this challenge is the first to mention Node.js and working with multiple JS files on a local machine. Maybe we could throw in a quick explanation of 'the why'? Something along the lines of:

> When working on large projects, it is easier to organise code when it is split between multiple files. This is where Node.js comes in. Node has many uses, which include the now-obsolete `require` method, and it's replacement, `import`.

**Comment by ezioda004** [comment]
@alanpaulprice Yes, I think its the first one to mention Node.
I don't think we need to add the reason why we use `require()` but rather than it should be about that it's a function provided by Node. That's it.

**Comment by alanpaulprice** [comment]
@ezioda004 Fair enough.

By the way, I just noticed that you said to discuss this on the issue, not here. My mistake!

**Comment by RandellDawson** [comment]
Closed and Reopened this PR to attempt to resolve a specific Travis build failure.

**Comment by RandellDawson** [comment]
@alanpaulprice @ezioda004 I noticed that the build is failing, because one of the tests fails:
```
  1 failing
  1) Check challenges tests
       Understand the Differences Between import and require
         Check tests against solutions
           Solution 1
             valid <code>import</code> statement:
     evalmachine.<anonymous>:9
import { capitalizeString } from "string_functions";
       ^
SyntaxError: Unexpected token {
      at new Script (vm.js:79:7)
      at runTestInJsdom (test/test-challenges.js:498:18)
      at evaluateJsTest (test/test-challenges.js:380:9)
      at Context.<anonymous> (test/test-challenges.js:177:25)
```

**Comment by ezioda004** [comment]
@RandellDawson The error I believe is due to the file not being of `"type = module"`, it throws an error. Omitting the solution is one way of fixing.

@ValeraS Is there another way to fix this?

**Comment by RandellDawson** [comment]
Maybe we can get @Bouncey to review and merge soon.  The only other PR with commits to test-challenges.js is PR #34376 and the change is not related.

**Comment by Bouncey** [comment]
Sorry, forgot I could click a button in Travis, build restarted

**Comment by ValeraS** [comment]
@Bouncey it does not work without #34449

**Comment by Bouncey** [comment]
I thought I had merged that. I will trigger another build when possible

**Comment by ValeraS** [comment]
@RandellDawson @ezioda004 Now tests work as they should

**Comment by RandellDawson** [comment]
@ValeraS I just tried clicking the Restart Build button and it still failed and then I closed/reopened the PR and it still fails.  What else needs to happen for this PR to pass the Travis requirements?

**Comment by RandellDawson** [comment]
@ValeraS OK.  That makes sense.  For some reason, I thought we were just waiting on that other PR to be merged, but now I understand the issue.

**Comment by alanpaulprice** [comment]
Oops, sorry! I missed the './' in the import statement. I've just made a commit adding it. It should work now!

**Review by ezioda004 — COMMENTED** [review]
><strong>Note</strong><br>The lessons in this section handle non-browser features. <code>import</code>, and the statements we introduce in the rest of these lessons, won't work on a browser directly. However, we can use various tools to create code out of this to make it work in browser.

Are we keeping this? As modern browsers natively support ES Modules.

**Comment by ezioda004** [comment]
@ValeraS The build is still failing (I guess because we dont have the import file). Is there a way to fake an import?

**Comment by ValeraS** [comment]
Look at Before test section

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/understand-the-differences-between-import-and-require.english.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Add the appropriate <code>import</code> statement that will allow the current fi
 ```yml
 tests:
   - text: valid <code>import</code> statement
```
Comment (thecodingaviator) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/understand-the-differences-between-import-and-require.english.md]:
```suggestion
  - text: The <code>import</code> statement should be valid
```

**Comment by thecodingaviator** [comment]
I've fixed the merge conflict @ezioda004 , the test is passing.

**Review by thecodingaviator — DISMISSED** [review]
@alanpaulprice Starting line 54, please fix the Before Test section to how it should look like before test

Edit: Ignore this

**Comment by RandellDawson** [comment]
After messing with this challenge again, I have noticed the only to use the import feature on the latest version of Chrome on Windows is to add the `.js` extension after "string_functions" or it will give the following error:
```
"Uncaught SyntaxError: Unexpected token {"
```
Below is how I can get it to work on Windows, so we will either need to add a note about how it is different on Windows or explain it in the description.  Otherwise, Windows users will assume they can take this example and directly apply it.
**index.html**
```html
<script type="module" src="index.js"></script> <!-- can also prefix like src="./index.js" -->
```

**string_functions.js**
```js
export const capitalizeString = str => str.toUpperCase();
```

**index.js**
```js
"use strict"; // this line is optional on Windows
import { capitalizeString } from './string_functions.js';
console.log(capitalizeString("hello!"));
```

I also think it might be important to add a note about the `type="module" part in the `index.html`.  This is not discussed in any detail in this challenge, but it will not work in all "modern" browsers without specifying it, unless you are using something like Webpack, which I assume is beyond the scope of this challenge.

**Comment by RandellDawson** [comment]
This PR has gone stale.

**Pull request closed without merging**